CMakeLists.txt

```diff
@@ -1,5 +1,8 @@
-cmake_minimum_required(VERSION 2.8.12.2)
-project(sqlitebrowser)
+cmake_minimum_required(VERSION 3.9)
+project(sqlitebrowser
+	VERSION 3.12.1
```

Comment:
Which benefit brings this?, because it's another place where the version has to be updated.

Reply:
It sets some CMake variables (`PROJECT_VERSION`, `PROJECT_VERSION_MAJOR` etc.) that are used e.g. by CPack. It's possible to read it from a file and set these variables manually, but before getting all the edge cases right I wanted to put up the PR for discussion.

Reply:
Ideally, it should be got from https://github.com/sqlitebrowser/sqlitebrowser/blob/master/src/version.h

Reply:
Or maybe the other way round?

CMake can generate files, so it could take the version specified here, split it and write it to `version.h` or pass the version information directly on the command line (`-D MAJOR_VERSION=3` etc.).

In an other project, I even have a few lines to detect if CMake is run from a git repository and add information (branch, commits since the last reachable tag) so the version number is more helpful in bug reports (think `v3.12.1+5_master_abc123def`).

Reply:
I've added a commit that generates the `version.h` based on the version defined in the `CMakeLists.txt`. It also creates a target for the main executable first (after settings `CMAKE_AUTOMOC`) so the compile definitions can be added with `target_compile_definitions`, the sources are added later with `target_sources`.

Reply:
> I've added a commit that generates the `version.h` based on the version defined in the `CMakeLists.txt`.

On one hand, that seems pretty smart. :smile:

But... won't that break the compile for people using qmake instead of CMake?

Reply:
> But... won't that break the compile for people using qmake instead of CMake?

Absolutely.

Qt itself is switching to CMake and CMake is a first class citizen for the offical Qt IDE (not to mention Visual Studio), so someone with more standing with  the project could suggest dropping support for qmake.

In case that's not an option / something for the distant future, you could just go ahead and cherry-pick the other commits.